src/osd/PrimaryLogPG.cc

```diff
@@ -10190,6 +10190,11 @@ int PrimaryLogPG::find_object_context(const hobject_t& oid,
     first = p->second.back();
     last = p->second.front();
   }
+  if (p->second.empty()) {
```

Comment:
@smithfarm this needs to be inside the above "else" block. Before `first = p->second.back()` -- we need to check that we can safely use `p->second.back()` and `p->second.front()`.

Reply:
@trociny Oh, right. Fixing.